hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2822,6 +2843,10 @@ func (r *HostedClusterReconciler) delete(ctx context.Context, req ctrl.Request,
 	if err := r.Delete(ctx, ns); err != nil && !apierrors.IsNotFound(err) {
 		return false, fmt.Errorf("failed to delete namespace: %w", err)
 	}
+
+	if err := r.cleanupOIDCBucketData(ctx, hc); err != nil {
```

Comment:
This call should be gated by
```
switch hcluster.Spec.Platform.Type { 
case hyperv1.AWSPlatform:
```
right?

Reply:
It internally gates by checking for the finalizer and doing nothing if it is absent